src/nupic/algorithms/AnomalyLikelihood.cpp

```diff
@@ -0,0 +1,592 @@
+  }
+
+ Real32 tailProbability(Real32 x) const {
+ /**
+  Given the normal distribution specified by the mean and standard deviation
+  in distributionParams, return the probability of getting samples further
+  from the mean. For values above the mean, this is the probability of getting
+  samples > x and for values below the mean, the probability of getting
+  samples < x. This is the Q-function: the tail probability of the normal distribution.
+
+  :param distributionParams: dict with 'mean' and 'stdev' of the distribution
+  **/
+     assert(distribution.name != "unknown" && distribution.stdev > 0);
+     
+  if (x < distribution.mean) {
+    // Gaussian is symmetrical around mean, so flip to get the tail probability
+    Real32 xp = 2 * distribution.mean - x;
+    assert(xp != x);
+    return tailProbability(xp);
+  }
+
+  // Calculate the Q function with the complementary error function, explained
+  // here: http://www.gaussianwaves.com/2012/07/q-function-and-error-functions
+  Real32 z = (x - distribution.mean) / distribution.stdev;
+  return 0.5 * erfc(z/1.4142);
+  }
+
+
+inline Real  computeLogLikelihood(Real likelihood) const {
+  /**
+    Compute a log scale representation of the likelihood value. Since the
+    likelihood computations return low probabilities that often go into four 9's
+    or five 9's, a log value is more useful for visualization, thresholding,
+    etc.
+   **/
+    // The log formula is:
+    //     Math.log(1.0000000001 - likelihood) / Math.log(1.0 - 0.9999999999)
+    return log(1.0000000001f - likelihood) / -23.02585084720009f;
+}
+
+
+DistributionParams estimateNormal(vector<Real> sampleData, bool performLowerBoundCheck=true) {
+  /** 
+  :param sampleData:
+  :type sampleData: Numpy array.
+  :param performLowerBoundCheck:
+  :type performLowerBoundCheck: bool
+  :returns: A dict containing the parameters of a normal distribution based on
+      the ``sampleData``.
+  **/
+    auto mean = compute_mean(sampleData);
+    auto var = compute_var(sampleData, mean);
+  DistributionParams params = DistributionParams("normal", mean, var, 0.0); 
+
+  if (performLowerBoundCheck) {
+    /* Handle edge case of almost no deviations and super low anomaly scores. We
+     find that such low anomaly means can happen, but then the slightest blip
+     of anomaly score can cause the likelihood to jump up to red.
+     */
+    if (params.mean < 0.03) {
+      params.mean= 0.03;
+    }
+
+    // Catch all for super low variance to handle numerical precision issues
+    if (params.variance < 0.0003) {
+      params.variance = 0.0003;
+    }
+  }
+
+  // Compute standard deviation
+  if (params.variance > 0) {
+    params.stdev = sqrt(params.variance);
+  } else{
+    params.stdev = 0;
+  }
+
+  return params;
+}
+
+Real compute_mean(vector<Real> v) const { //TODO do we have a (more comp. stable) implementation of mean/variance?
+    Real sum = std::accumulate(v.begin(), v.end(), 0.0);
+    return sum / v.size();
```

Comment:
TODO is there an implementation of mean/variance to reuse? instead of writing it here...